src/main/java/io/fabric8/maven/docker/util/EnvUtil.java

```diff
@@ -78,34 +81,54 @@ public static boolean greaterOrEqualsVersion(String versionA, String versionB) {
         return largerVersion != null && largerVersion.equals(versionA);
     }
 
+    private static final Function<String, String[]> SPLIT_ON_LAST_COLON = new Function<String, String[]>() {
+        @Override
+        public String[] apply(String element) {
+          int colon = element.lastIndexOf(':');
+          if (colon < 0) {
+              return new String[] {element, element};
+          } else {
+              return new String[] {element.substring(0, colon), element.substring(colon + 1)};
+          }
+        }
+    };
+
     /**
      * Splits every element in the given list on the last colon in the name and returns a list with
-     * two elements: The left part before the colon and the right part after the colon. If the string doesnt contain
-     * a colon, the value is used for both elements in the returned arrays.
+     * two elements: The left part before the colon and the right part after the colon. If the string
+     * doesn't contain a colon, the value is used for both elements in the returned arrays.
      *
      * @param listToSplit list of strings to split
      * @return return list of 2-element arrays or an empty list if the given list is empty or null
      */
-    public static List<String[]> splitOnLastColon(List<String> listToSplit) {
+    public static Iterable<String[]> splitOnLastColon(Iterable<String> listToSplit) {
         if (listToSplit != null) {
-            List<String[]> ret = new ArrayList<>();
-
-            for (String element : listToSplit) {
-                String[] p = element.split(":");
-                String rightValue = p[p.length - 1];
-                String[] nameParts = Arrays.copyOfRange(p, 0, p.length - 1);
-                String leftValue = StringUtils.join(nameParts, ":");
-                if (leftValue.length() == 0) {
-                    leftValue = rightValue;
-                }
-                ret.add(new String[]{leftValue, rightValue});
```

Comment:
Oops, that's really a unneeded complicated algo. good that its simplified now :)